manifests/job/present.pp

```diff
@@ -29,8 +33,19 @@
   }
 
   $jenkins_cli        = $jenkins::cli::cmd
-  $tmp_config_path    = "/tmp/${jobname}-config.xml"
-  $job_dir            = "${::jenkins::job_dir}/${jobname}"
+
+  # Add support for Folders plugin
+  if $folder {
+    $realjobname     = "${folder}/${jobname}"
+    $job_dir         = "${::jenkins::job_dir}/${folder}/jobs/${jobname}"
+    $tmp_config_path = "/tmp/${folder}/${jobname}-config.xml"
```

Comment:
Does it make sense to make `$folder` part of the temp file name instead of creating a subdir?  That would eliminate the need for an additional file resource.